service/frontend/workflow_handler.go

```diff
@@ -2743,27 +2747,59 @@ func (wh *WorkflowHandler) DescribeWorkflowExecution(ctx context.Context, reques
 		return nil, err
 	}
 
-	response, err := wh.historyClient.DescribeWorkflowExecution(ctx, &historyservice.DescribeWorkflowExecutionRequest{
-		NamespaceId: namespaceID.String(),
-		Request:     request,
-	})
+	response, err := wh.historyClient.DescribeWorkflowExecution(
+		ctx,
+		&historyservice.DescribeWorkflowExecutionRequest{
+			NamespaceId: namespaceID.String(),
+			Request:     request,
+		},
+	)
 
 	if err != nil {
 		return nil, err
 	}
 
-	if response.GetWorkflowExecutionInfo().GetSearchAttributes() != nil {
-		saTypeMap, err := wh.saProvider.GetSearchAttributes(wh.config.ESIndexName, false)
-		if err != nil {
-			return nil, serviceerror.NewUnavailable(fmt.Sprintf(errUnableToGetSearchAttributesMessage, err))
-		}
-		searchattribute.ApplyTypeMap(response.GetWorkflowExecutionInfo().GetSearchAttributes(), saTypeMap)
-		aliasedSas, err := searchattribute.AliasFields(wh.saMapper, response.GetWorkflowExecutionInfo().GetSearchAttributes(), request.GetNamespace())
-		if err != nil {
-			return nil, err
+	executionInfo := response.GetWorkflowExecutionInfo()
+	if executionInfo != nil {
```

Comment:
nit: Shall we return an error if executionInfo is nil?

Reply:
If `historyClient.DescribeWorkflowExecution` didn't return an error, this should always be not nil. I don't think the check is necessary actually.